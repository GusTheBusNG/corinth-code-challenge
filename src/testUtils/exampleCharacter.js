export const character = {
  name: 'Anakin Skywalker',
  height: '6',
  mass: 'something',
  hair_color: 'brown',
  birth_year: 'some time long ago',
  species: ['example url species'],
  starships: ['example url starships'],
  films: ['example url fil1s']
}
