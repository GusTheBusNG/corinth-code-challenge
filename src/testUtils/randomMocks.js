export const stubLoading = (message = 'loading') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getLoadingQuote = jest.fn(() => message)
}

export const stubError = (message = 'error') => {
  const random = jest.requireActual('../utils/randomPhrases');
  random.getErrorQuote = jest.fn(() => message)
}