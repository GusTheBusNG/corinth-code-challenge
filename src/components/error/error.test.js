import React from 'react';
import { create } from 'react-test-renderer';

import Error from './';
import { stubError } from '../../testUtils/randomMocks'

const createError = () => create(<Error error='error' />);

describe('Error', () => {
  describe('render', () => {
    it('renders correctly', async () => {
      stubError()

      expect(createError().toJSON()).toMatchSnapshot();
    })
  })
})
