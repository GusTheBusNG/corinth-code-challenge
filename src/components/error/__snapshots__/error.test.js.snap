// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error render renders correctly 1`] = `
<p
  className="error"
>
  error
</p>
`;
